docs/guides/api-graphql/fragments/android/subscriptions-by-id.md

```diff
@@ -0,0 +1,124 @@
+Now you can create a custom subscription for comment creation with a specific post id:
+
+<amplify-block-switcher>
+<amplify-block name="Java">
+
+```java
+private GraphQLRequest<Comment> onCreateCommentByPostId(String id) {
+    String document = "subscription onCreateCommentByPostId($id: ID!) { "
+        + "onCommentByPostId(postCommentsId: $id) { "
+            + "content "
+            + "id "
+            + "postCommentsId "
+        + "}"
+    + "}";
+    return new SimpleGraphQLRequest<>(
+            document,
+            Collections.singletonMap("id", id),
+            Comment.class,
+            new GsonVariablesSerializer());
+}
+```
+
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+private fun onCreateCommentByPostId(id: String): GraphQLRequest<Comment?>? {
+    val document = ("subscription onCreateCommentByPostId(\$id: ID!) { "
+        + "onCommentByPostId(postCommentsId: \$id) { "
+            + "content "
+            + "id "
+            + "postCommentsId "
+        + "}"
+    + "}")
+    return SimpleGraphQLRequest(
+            document,
+            mapOf("id", id),
+            Comment::class.java,
+            GsonVariablesSerializer())
+}
```

Comment:
Suggest three tweaks:
1. Use +4 indentation instead of +8, to help maximize use of constrained view ports.
2. `mapOf` had syntax error, should be `mapOf("id" to id)`
3. Use a `"""` multi-line comment for the GraphQL document, and `${'$'}` as the escape sequence for a literal `$` character.

Question: can we remove either (or both) of the `?` in `GraphQLRequest<Comment?>?` ?
```suggestion
private fun onCreateCommentByPostId(id: String): GraphQLRequest<Comment?>? {
    val document = """
        subscription onCreateCommentByPostId(${'$'}id: ID!) {
            onCommentByPostId(postCommentsId: ${'$'}id) {
                content
                id
                postCommentsId
            }
         }
    """.trimIndent()
    return SimpleGraphQLRequest(
        document,
        mapOf("id" to id),
        Comment::class.java,
        GsonVariablesSerializer()
    )
}
```

Reply:
Thanks for the suggestions!   TBH, I didn't do much more than what the Kotlin auto converter did for me :). Multi-line string literals are great though.  It really makes me want to use Kotlin more. 

I just removed the two `?` as well, so now the return type is `GraphQLRequest<Comment>`.  The question marks should only be needed if the return type might be null, which it won't be.